service/test/agama/config_test.rb

```diff
@@ -91,17 +91,25 @@
 
   describe "#products" do
     it "returns products available for current hardware" do
-      subject = described_class.from_file(File.join(FIXTURES_PATH, "agama-archs.yaml"))
-      expect(subject.products.size).to eq 2
+      allow_any_instance_of(Agama::ProductReader).to receive(:load_products).and_return([
+          {
+            "id" => "test",
+            "archs" => "x86_64"
+          },
+          {
+            "id" => "test2",
+            "archs" => "s390x"
+          }
+        ])
+      expect(Yast2::ArchFilter).to receive(:from_string).twice.and_return(double(match?: true), double(match?: false))
+      expect(subject.products.size).to eq 1
     end
   end
 
   describe "#multi_product?" do
     context "when more than one product is defined" do
-      subject do
-        described_class.from_file(
-          File.join(FIXTURES_PATH, "root_dir", "etc", "agama.yaml")
-        )
+      before do
+        allow_any_instance_of(Agama::ProductReader).to receive(:load_products).and_call_original
```

Comment:
I am not a big fan of `allow_any_instance_of`. But, as it is a small test, I guess it is fine.

Reply:
I can change it to new and return double.

Reply:
I use same construct in global test_helper, so it is not just small test